### ISSUE TITLE
about SSR

### ISSUE THREAD

**Issue by Ariex** [andreiduca/cra-ssr-code-splitting#3]
not sure if this is called "SSR", because neither "Hi, I'm async." or "Redux: Hi, I'm from client!" are not rendered at server side. By looking at the page source (which is what a bot get normally), there is no SEO value at all. Is this because I opened it in a wrong way?
Thanks.

**Comment by Ariex** [comment]
seems it is I open it in a wrong way. This only happens in my chrome, FF and IE are all gives me expected source code. Will close it now, hope can help with others having same issues in future.

Thanks very much for the article!

**Comment by andreiduca** [comment]
The idea of SSR is to render *at least* something on the server, any part of the entire React app that you think it's useful. The different messages `I'm from client` vs `I'm from server` are just to illustrate where those messages were generated.

So: if SSR is successful, then you should see the server-generated message (both in browser and in the HTML source). If it's not, then you would get an empty HTML, and only after the JS is loaded, the React app is rendered client-side, and (only after the components are mounted) you see the client-generated message.

Please note: the client-generated message should never be rendered on the server!

In conclusion, I believe this is a non-issue. Hope I made it more clear about what's happening.

**Comment by Ariex** [comment]
yep, there is no issue in your code, all working perfectly.
It is my chrome some how cached something, which keep gives me wrong result. I believe it is the service worker. Since I followed your article, so it somehow cached the earlier version of the html page, and keep server me that one. And this issue is still there even I switch to incognito mode. I only realized this when i decide give FF a try.

I would like to say again, there is absolute *NO* problem in your article or code, it is just my chrome. I am sorry if there are any confusion in my previous comments.
